backend/app/api/ws/chat_namespace.py

```
@@ -1565,20 +1353,38 @@ async def on_chat_retry(self, sid: str, data: dict) -> dict:
+        try:
+            await trigger_ai_response_unified(
+                namespace=self,
+                **dispatch_args_or_error,
+            )
+        except Exception as e:
+            from sqlalchemy.exc import SQLAlchemyError
+
+            logger.error(f"[WS] chat:retry exception: {e}", exc_info=True)
             if isinstance(e, SQLAlchemyError):
                 return {"error": "Database error occurred"}
             return {"error": f"Internal server error: {str(e)}"}
```

Ruff (0.15.15)

[warning] Use explicit conversion flag

Replace with conversion flag

(RUF010)